### PR TITLE
Fix #3975: Tab Tray refactor.

### LINKED ISSUE #3975: Tab Tray collection view crash.
### Steps to Reproduce 
Hard to reproduce, sometimes it seems to crash when a favorite is removed or reordered

**Actual result:** 

**Expected result:**

**Reproduces how often:** [Easily reproduced, Intermittent Issue]

**Brave Version:** 

**Device details:** 

**Website problems only:**
- did you check with Brave Shields down?
- did you check in Safari/Firefox (WkWebView-based browsers)?

### PR DESCRIPTION
**Pull request by iccub**
## Summary of Changes

This pull request fixes #3975 

## Submitter Checklist:

- [x] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`

## Test Plan:

Verify that tab tray actions work with the new implementation, test stuff like:
- switching between private and regular modes
- add new tab, remove tab
- reorder tabs
- remove tab by swiping left or right

## Reviewer Checklist:

- [x] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `bug` / `enhancement`
- [x] Necessary [security reviews](https://github.com/brave/security/issues/new/choose) have taken place.
- [x] Adequate unit test coverage exists to prevent regressions.
- [x] Adequate test plan exists for QA to validate (if applicable).
- [x] Issue and pull request is assigned to a milestone (should happen at merge time).

### CODE REVIEW

File: Client/Frontend/Browser/Tab Tray/TabTrayController2.swift [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+import Shared
+import BraveShared
+
+protocol TabTrayDelegate: AnyObject {
+    /// Notifies the delegate that order of tabs on tab tray has changed.
+    /// This info can be used to update UI in other place, for example update order of tabs in tabs bar.
+    func tabOrderChanged()
+}
+
+class TabTrayController2: UIViewController {
```
Comment (iccub) [inline Client/Frontend/Browser/Tab Tray/TabTrayController2.swift]: Once approved is ready i will rename it to `TabTrayController`, this is new file so did not want it to mess with the old tab tray vc class

File: Client/Frontend/Browser/Tab Tray/TabTrayController2.swift [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+        }
+        
+        return .init(operation: .move, intent: .insertAtDestinationIndexPath)
+    }
+}
+
+// MARK: - UIScrollViewAccessibilityDelegate
+
+extension TabTrayController2: UIScrollViewAccessibilityDelegate {
+    func accessibilityScrollStatus(for scrollView: UIScrollView) -> String? {
```
Comment (iccub) [inline Client/Frontend/Browser/Tab Tray/TabTrayController2.swift]: This implementation is 100% taken from the old class

File: Client/Frontend/Browser/Tab Tray/TabTrayView.swift [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import UIKit
+import Shared
+import BraveShared
+import BraveUI
+
+extension TabTrayController2 {
+    
+    class View: UIView {
```
Comment (iccub) [inline Client/Frontend/Browser/Tab Tray/TabTrayView.swift:13]: this is old implementation as well

**Comment by Brandon-T** [comment]
1. Open NTP.
2. Tap on Favourites (Wikipedia).
3. BEFORE it finishes loading, open another tab quickly.
4. Open Tab-Tray.
5. Screenshot is blank for Wikipedia.
6. Close Tab-Tray and go to Wikipedia Tab.
7. Wikipedia finishes loading.
8. Switch to the New Tab.
9. Open Tab-Tray.
10. Screenshot is still blank (it didn't update).

If you open Tab-Tray on wikipedia tab, screenshot updates just fine.

**Review by Brandon-T — CHANGES_REQUESTED** [review]
Crashes when opening multiple tabs (more than 3.. IE: 4, 5).
Switching tabs quickly seems to cause it. 
Before it finishes switching the tab, open tab-tray.
**Info sent in Slack.**

**Review by Brandon-T — APPROVED** [review]

**Pull request merged**